src/FormParserFactory.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace React\Http;
+
+use React\Http\Parser\FormUrlencoded;
+use React\Http\Parser\Multipart;
+
+class FormParserFactory
+{
+    public static function create(Request $request)
+    {
+        $headers = $request->getHeaders();
+        $headers = array_change_key_case($headers, CASE_LOWER);
+
+        if (!array_key_exists('content-type', $headers)) {
```

Comment:
`Content-Type` is optional in http spec, so we should really not throw an exception, because this it's not a exception. Better way is to treat it as if there is no body at all. However, this is already the case. All these comments (above and mine) refer to an old commit (d525755623ae8a2776ff196bae0d0523ebb77087)
See current: https://github.com/reactphp/http/pull/41/files#diff-5f9f68b3995267f69b2f776127df04f0